x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/AbstractStepMasterTimeoutTestCase.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.MasterNodeRequest;
+import org.elasticsearch.client.AdminClient;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.IndicesAdminClient;
+import org.elasticsearch.cluster.ClusterName;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.test.client.NoOpClient;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.junit.After;
+import org.junit.Before;
+import org.mockito.Mockito;
+
+import static org.elasticsearch.xpack.core.ilm.LifecycleSettings.LIFECYCLE_STEP_MASTER_TIMEOUT;
+import static org.hamcrest.Matchers.equalTo;
+
+public abstract class AbstractStepMasterTimeoutTestCase<T extends AsyncActionStep> extends AbstractStepTestCase<T> {
+
+    protected Client client;
+    protected AdminClient adminClient;
+    protected IndicesAdminClient indicesClient;
+    protected ThreadPool pool;
+
+    @Before
+    public void setup() {
+        client = Mockito.mock(Client.class);
```

Comment:
Most subclasses already shadow these mockings/instances in the tests where they are applicable. I think this base class should not attempt to instantiate more than it uses/templates (ie. `ThreadPool`) and if we decide to centralise all the mocks/stubs in the base class I think we should remove the re-declaration from the tests (eg. OpenFollowerIndexStepTests, RolloverStepTests etc). I think that could be a follow-up PR to reduce the scope of this one, so I propose we remove the mocks/stubs from the base class for now. 

What do you think?

Reply:
That makes sense. This change was mostly due to the fact that I needed to overwrite client used in tests from `AbstractStepMasterTimeoutTestCase` - client was passed on instance creation in `createRandomInstance`. For now on I reverted client initialization but added visible-for-testing method `AsyncActionStep#setClient` that lets you overwrite client after creation.
I'll follow up soon with PR for centralizing this initialization and this method can be removed then.

Reply:
Thanks for making the change @probakowski 

I think now it's at least explicit (I found it rather confusing to see what client is used where before) but I agree it'll be nice to centralise the initialization process.